core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1102,6 +1121,22 @@ public Builder setDefaultPartitionSpec(int specId) {
       return this;
     }
 
+    private Builder removePartitionSpec(PartitionSpec spec) {
+      Preconditions.checkArgument(
+          changes.isEmpty(), "Cannot remove partition spec with other metadata update");
```

Comment:
Why is this necessary?

Reply:
I think this is needed to avoid conflicts with changes to the default spec ID. I'd probably change this to allow other changes and instead update this and the methods that set the default spec ID to check for one another.